src/Core/Checkout/Cart/Delivery/DeliveryProcessor.php

```diff
@@ -93,18 +96,43 @@ public function process(CartDataCollection $data, Cart $original, Cart $toCalcul
                     return $delivery->getShippingCosts()->getTotalPrice() >= 0;
                 });
 
-                $firstDelivery = $deliveries->first();
-                if ($firstDelivery === null) {
+                if (!Feature::isActive('v6.8.0.0')) {
```

Comment:
instead of duplicating so much code, can't you just assign the primary vs the first delivery to the same variable and then execute the same code?